### ISSUE TITLE
Cannot execute a vim command from a new asyncio task

### ISSUE THREAD

**Issue by tbodt** [neovim/python-client#342]

**Comment by blueyed** [comment]
Just tried it, and it works for me using Python 3.6.6 and pynvim master.
It needs `\n` to actually write it though.

Also `vim.command("echom 1")` worked.

**Comment by blueyed** [comment]
However, only the first command gets run.  With the following only `1` will be displayed:

```vim
:py3 async def foo(): vim.command('echom 1'); vim.command('echom 2')
:py3 import asyncio; asyncio.ensure_future(foo())
```

**Comment by blueyed** [comment]
Yeah, your issue after all.. - so confirming ;)

**Comment by blueyed** [comment]
https://github.com/neovim/python-client/blob/fbf0c843284ee33eca2dc4d62aec178809b0463d/neovim/msgpack_rpc/session.py#L63-L69

**Comment by bfredl** [comment]
nvim-managed greenlets and asyncio coroutines are like water and oil, they are incompatible event/async primitives that simply doesn't mix: any attempt to "seamlessly" integrate them might fail in the first request stack/event handing situation that wasn't explicitly considered.

What we could add is (1) a greenlet-callable function, that runs an asyncio coro/future until completion and then resumes the greenlet and (2) an awaitable function that suspends a coroutine and runs a closure as a greenlet until it completes and then resumes the coroutine. Though this will more or less be callback hell and thus have none of the benefits of _either_ the stackful greenlet approach or the explicitly yieldable approach of asyncio coroutines. 

Rather what I would imagine useful is to reimplement the python client using _only_ asyncio and native coroutines and no greenlets, I suspect that will be a lot simpler than the exisiting multi-layer client: it will more or less be a library providing a `nvim` asyncio protocol for its native transports, and request awaitables for coroutines which asyncio otherwise manages by itself.

**Comment by tbodt** [comment]
I would really love to have all these strange event loops dropped and replace with asyncio/trollius, though wouldn't there be a backwards compatibility problem?

**Comment by bfredl** [comment]
> though wouldn't there be a backwards compatibility problem?

Yes. It would need be a separate project from this. I wouldn't even think "fork", rather new codebase which can happen to borrow code from this where it makes sense. Starting from scratch with no regards of vim compatibility (which necessitates a sync-like interface backed by greenlets or similar) would give a lot more freedom for both simplicity and expressivity. For instance `call_atomic` could be exposed as a function converting multiple request awaitables to a single awaitable, while with this codebase to use `call_atomic` you must throw all existing abstractions above the raw API out through the window.

**Comment by Shougo** [comment]
I need the fix for denite.nvim split processes feature.

https://github.com/Shougo/denite.nvim/issues/509
